maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java

```diff
@@ -270,6 +277,137 @@ public String promptForPlatformVersion(List<String> versions) {
+    public String promptForJdkPath(Server server) {
+
+        List<String> paths = new ArrayList<>();
+        paths.add("JAVA_HOME (currently: " + System.getProperty("java.home") + ")");
+        try {
+            paths.addAll(getJdkPathsFromProperties(getSdkProperties()));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        if (!interactiveMode) {
+            return paths.get(0);
+        }
+
+        String path = promptForMissingValueWithOptions(SDKConstants.OPENMRS_SDK_JDK_OPTION,
+                server.getJdkPath(), "path", paths, SDKConstants.OPENMRS_SDK_JDK_CUSTOM, null);
+
+        if (path.equals(paths.get(0))) {
+            server.setJdkPath(System.getProperty("java.home"));
+        }
+
+        if (!isThereJdkUnderPath(path)) {
+                System.out.println(SDKConstants.OPENMRS_SDK_JDK_CUSTOM_INVALID);
+                server.setJdkPath(promptForJdkPath(server));
+        }
+        else {
+            if (!isPathLineDuplicated(paths, path)) {
+                try {
+                    addJdkPathToSdkProperties(path);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+            return path;
+        }
+        return null;
+    }
+
+    private void addJdkPathToSdkProperties(String path) throws IOException {
+
+        File sdkPropertiesFile = new File(Server.getServersPathFile(), SDKConstants.OPENMRS_SDK_PROPERTIES);
+        Properties sdkProperties = getSdkProperties();
+        List<String> jdkPaths = getJdkPathsFromProperties(sdkProperties);
+
+        if (!path.equals(System.getProperty("java.home")) && !isPathLineDuplicated(jdkPaths, path)) {
+            if (jdkPaths.size() == 5) {
+                jdkPaths.set(4, path);
+            }
+            else {
+                jdkPaths.add(path);
+            }
+
+            Collections.sort(jdkPaths);
+
+            String updatedProperty = convertListToPropertyString(jdkPaths);
+            sdkProperties.setProperty(SDKConstants.OPENMRS_SDK_PROPERTIES_JDK_HOME, updatedProperty);
+            savePropertiesChangesToFile(sdkProperties, sdkPropertiesFile, AVAILABLE_JDK_VERSIONS);
+        }
+    }
+
+    private String convertListToPropertyString(List<String> list) {
+        String result = "";
+        for (String str : list) {
+            result += str + ", ";
+        }
+        return result;
+    }
+
+    public Properties getSdkProperties() throws IOException {
+        File sdkPropertiesFile = new File(Server.getServersPathFile(), SDKConstants.OPENMRS_SDK_PROPERTIES);
+
+        if (!sdkPropertiesFile.exists()) {
+            sdkPropertiesFile.createNewFile();
+        }
+
+        InputStream in = new FileInputStream(sdkPropertiesFile);
+        Properties sdkProperties = new Properties();
+        sdkProperties.load(in);
+        return sdkProperties;
+    }
+
+    public ArrayList<String> getJdkPathsFromProperties(Properties sdkProperties) {
```

Comment:
I won't get rid of getSdkProperties() since its preventing code repeat, but I've changed getJdkPathsFromProperties(Properties sdkProperties) to getJdkPaths() (im getting properties from getSdkProperties() method inside getJdkPaths(), so i don't need to pass it through argument anymore).